homeassistant/core.py

```diff
@@ -472,7 +472,8 @@ def set(self, entity_id, new_state, attributes=None):
             old_state = self._states.get(entity_id)
 
             is_existing = old_state is not None
-            same_state = is_existing and old_state.state == new_state
+            same_state = (is_existing and old_state.state == new_state and
+                          not force_update)
```

Comment:
If an update is forced, should the the last_changed value be updated or should it stay the same? Currently, I have last_changed update. That's why I have the check where it is, it affects two statements: the if-statement below and the last_changed assignment.